Content.Server/Chemistry/ReagentEffects/MakeSentient.cs

```diff
@@ -16,25 +16,20 @@ public override void Effect(ReagentEffectArgs args)
         var entityManager = args.EntityManager;
         var uid = args.SolutionEntity;
 
-        // This makes it so it doesn't affect things that are already sentient
-        if (entityManager.HasComponent<MindContainerComponent>(uid))
-        {
-            return;
-        }
-
-        // This piece of code makes things able to speak "normally". One thing of note is that monkeys have a unique accent and won't be affected by this.
+        // Let affected entities speak normally to make this effect different from, say, the "random sentience" event
+        // This also works on entities that already have a mind
+        // We call this before the mind check to allow things like player-controlled mice to be able to benefit from the effect
         entityManager.RemoveComponent<ReplacementAccentComponent>(uid);
-
-        // Monke talk. This makes cognizine a cure to AMIV's long term damage funnily enough, do with this information what you will.
         entityManager.RemoveComponent<MonkeyAccentComponent>(uid);
 
-        // This makes it so it doesn't add a ghost role to things that are already sentient
+        // Stops from adding a ghost role to things like people who already have a mind
         if (entityManager.HasComponent<MindContainerComponent>(uid))
         {
             return;
         }
 
-        // No idea what anything past this point does
+        // Stops from adding a ghost role to things that already have ghost roles
+        // The check for GhostTakeoverAvailableComponent may be redundant but it's hard to tell
```

Comment:
It is not redundant, because `GhostRoleComponent`s can be used with either `GhostTakeoverAvailableComponent`s or `GhostRoleMobSpawnerComponent`s.

Reply:
Ah, got it

Reply:
Although... in either situation, there's still a `GhostRoleComponent`, and there's no way that an entity that hasn't spawned yet could get injected with cognizine, so unless I'm missing something it probably _is_ redundant.

Reply:
Testing and this new information you've given me showed that the GhostTakeoverAvailableComponent check was indeed redundant. It and the GhostTakeoverAvailableComponent removal line on probability fail have both been removed, and a line got changed to EnsureComp (see PR description).